kbhff/api/email.py

```diff
@@ -1,6 +1,36 @@
 import easyimap, email
 from kbhff.api.exceptions import *
 import time
+import imaplib
+import sys
+
+class GmailConnection:
+    def __init__(self):
+        self.connection = None
+        self.mail_credentials = get_mail_credentials()
+        self.connect()
+
+    def __del__(self):
+        if self.connection is not None:
+            self.connection.quit()
+    
+    def connect(self):
+        if self.connection is not None:
+            self.connection.quit()
+        address = self.mail_credentials["login"]
+        password = self.mail_credentials["password"]
+        self.connection = easyimap.connect('imap.gmail.com', address, password)
+
+    def listup(self, number_of_mails = 25, retries_left = 100):
```

Comment:
I think this is more legible as a `for` loop instead of as a recursive function.